File: androidApp/src/main/kotlin/com/slowbuild/storyverse/storyverse/ui/common/StoryVerseTopBar.kt
package com.slowbuild.storyverse.storyverse.ui.common

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.RowScope
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.slowbuild.storyverse.storyverse.theme.LocalStoryVerseColors
import com.slowbuild.storyverse.storyverse.theme.toComposeColor

@Composable
fun StoryVerseTopBar(
    title: String,
    modifier: Modifier = Modifier,
    canNavigateBack: Boolean = false,
    onNavigateBack: () -> Unit = {},
    actions: @Composable RowScope.() -> Unit = {}
) {
    val themeColors = LocalStoryVerseColors.current
    val backgroundColor = themeColors.background.toComposeColor()
    val contentColor = themeColors.onBackground.toComposeColor()

    Row(
        modifier = modifier
            .fillMaxWidth()
            .height(56.dp)
            .background(backgroundColor)
            .padding(horizontal = 4.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        if (canNavigateBack) {
            IconButton(onClick = onNavigateBack) {
                Icon(
                    imageVector = Icons.AutoMirrored.Filled.ArrowBack,
                    contentDescription = "Back",
                    tint = contentColor
                )
            }
        } else {
            Spacer(modifier = Modifier.width(12.dp))
        }

        Text(
            text = title,
            fontSize = 19.sp,
            fontWeight = FontWeight.Bold,
            maxLines = 1,
            overflow = TextOverflow.Ellipsis,
            color = contentColor,
            modifier = Modifier.weight(1f)
        )

        Row(
            verticalAlignment = Alignment.CenterVertically,
            content = actions
        )
    }
}

File: iosApp/iosApp/Components/StoryVerseTopBar.swift
import SwiftUI
import SharedLogic

public struct StoryVerseTopBar<Actions: View>: View {
    @EnvironmentObject var themeManager: ThemeManager

    public let title: String
    public let canNavigateBack: Bool
    public let onNavigateBack: (() -> Void)?
    public let actions: Actions

    public init(
        title: String,
        canNavigateBack: Bool = false,
        onNavigateBack: (() -> Void)? = nil,
        @ViewBuilder actions: () -> Actions = { SwiftUI.EmptyView() }
    ) {
        self.title = title
        self.canNavigateBack = canNavigateBack
        self.onNavigateBack = onNavigateBack
        self.actions = actions()
    }

    public var body: some View {
        HStack(spacing: 12) {
            if canNavigateBack {
                Button(action: {
                    onNavigateBack?()
                }) {
                    Image(systemName: "arrow.backward")
                        .font(.system(size: 18, weight: .semibold))
                        .foregroundColor(themeManager.colors.onBackground)
                        .frame(width: 36, height: 36)
                }
            }

            Text(title)
                .font(.system(size: 19, weight: .bold))
                .foregroundColor(themeManager.colors.onBackground)
                .lineLimit(1)

            Spacer()

            actions
        }
        .padding(.horizontal, 16)
        .frame(height: 52)
        .background(themeManager.colors.background)
    }
}
